core/services/keystore/keys/ethkey/models.go

```diff
@@ -6,19 +6,26 @@ import (
 	"github.com/smartcontractkit/chainlink/core/utils"
 )
 
-type State struct {
-	ID         int32
-	Address    EIP55Address
-	EVMChainID utils.Big
+type KeyState[ADDR any, ID any, META any] struct {
```

Comment:
So the TxMgr doesn't actually use all the fields in this struct.
It only uses 2:
- Address
- Disabled

The Disabled field too, it just uses to filter out keys that are not enabled.
So literally all that TxMgr cares about, is to get a list of enabled Addresses.

We wouldn't want to expose everything when we don't even need all of this.

Lets not touch this struct here.
Instead see my suggestion on the KeyStore interface